roachtest: restore2TB/nodes=32 failed

roachtest.restore2TB/nodes=32 [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4299152&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4299152&tab=artifacts#/restore2TB/nodes=32) on master @ [8548987813ff9e1b8a9878023d3abfc6911c16db](https://github.com/cockroachdb/cockroach/commits/8548987813ff9e1b8a9878023d3abfc6911c16db):


```
		Wraps: (2) output in run_102003.382505255_n1_cockroach_sql
		Wraps: (3) ./cockroach sql --insecure -e "
		  | 				RESTORE csv.bank FROM
		  | 				'gs://cockroach-fixtures/workload/bank/version=1.0.0,payload-bytes=10240,ranges=0,rows=65104166,seed=1/bank?AUTH=implicit'
		  | 				WITH into_db = 'restore2tb'" returned
		  | stderr:
		  | ERROR: -e: driver: bad connection
		  | Failed running "sql"
		  |
		  | stdout:
		Wraps: (4) COMMAND_PROBLEM
		Wraps: (5) Node 1. Command with error:
		  | ``````
		  | ./cockroach sql --insecure -e "
		  | 				RESTORE csv.bank FROM
		  | 				'gs://cockroach-fixtures/workload/bank/version=1.0.0,payload-bytes=10240,ranges=0,rows=65104166,seed=1/bank?AUTH=implicit'
		  | 				WITH into_db = 'restore2tb'"
		  | ``````
		Wraps: (6) exit status 1
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) errors.Cmd (5) *hintdetail.withDetail (6) *exec.ExitError

	monitor.go:127,restore.go:454,test_runner.go:780: monitor failure: monitor task failed: t.Fatal() was called
		(1) attached stack trace
		  -- stack trace:
		  | main.(*monitorImpl).WaitE
		  | 	main/pkg/cmd/roachtest/monitor.go:115
		  | main.(*monitorImpl).Wait
		  | 	main/pkg/cmd/roachtest/monitor.go:123
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerRestore.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/restore.go:454
		  | main.(*testRunner).runTest.func2
		  | 	main/pkg/cmd/roachtest/test_runner.go:780
		Wraps: (2) monitor failure
		Wraps: (3) attached stack trace
		  -- stack trace:
		  | main.(*monitorImpl).wait.func2
		  | 	main/pkg/cmd/roachtest/monitor.go:171
		Wraps: (4) monitor task failed
		Wraps: (5) attached stack trace
		  -- stack trace:
		  | main.init
		  | 	main/pkg/cmd/roachtest/monitor.go:80
		  | runtime.doInit
		  | 	GOROOT/src/runtime/proc.go:6498
		  | runtime.main
		  | 	GOROOT/src/runtime/proc.go:238
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1581
		Wraps: (6) t.Fatal() was called
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/bulk-io
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*restore2TB/nodes=32.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

bingo:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x16bdd4d]

goroutine 17204 [running]:
panic({0x45ace20, 0x8c53790})
        GOROOT/src/runtime/panic.go:1147 +0x3a8 fp=0xc0045ce418 sp=0xc0045ce358 pc=0x48a768
runtime.panicmem(...)
        GOROOT/src/runtime/panic.go:221
runtime.sigpanic()
        GOROOT/src/runtime/signal_unix.go:735 +0x327 fp=0xc0045ce468 sp=0xc0045ce418 pc=0x4a1187
github.com/cockroachdb/cockroach/pkg/sql/catalog/dbdesc.maybeConvertIncompatibleDBPrivilegesToDefaultPrivileges(0x60ab770, 0xc00ac76380)
        github.com/cockroachdb/cockroach/pkg/sql/catalog/dbdesc/database_desc_builder.go:103 +0x4d fp=0xc0045ce520 sp=0xc0045ce468 pc=0x16bdd4d
github.com/cockroachdb/cockroach/pkg/sql/catalog/dbdesc.(*databaseDescriptorBuilder).RunPostDeserializationChanges(0xc008b8e7f8)
        github.com/cockroachdb/cockroach/pkg/sql/catalog/dbdesc/database_desc_builder.go:72 +0xd9 fp=0xc0045ce578 sp=0xc0045ce520 pc=0x16bdb99
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.maybeUpgradeDescriptors({0xc0030bc420, 0x2, 0x2}, 0x0)
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl/pkg/ccl/backupccl/restore_planning.go:997 +0x143 fp=0xc0045ce5f0 sp=0xc0045ce578 pc=0x3546c83
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.doRestorePlan({0x6043f38, 0xc00a193b60}, 0xc003e1ca00, {0x6148c80, 0xc001e45278}, {0xc00a18e4e0, 0x1, 0x1}, {0x0, 0x0, ...}, ...)
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl/pkg/ccl/backupccl/restore_planning.go:1959 +0x1132 fp=0xc0045cfcf8 sp=0xc0045ce5f0 pc=0x354cf32
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.restorePlanHook.func2({0x6043f38, 0xc00a193ad0}, {0x0, 0x33d75e0, 0xc0038327ac}, 0xc00a18e348)
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl/pkg/ccl/backupccl/restore_planning.go:1664 +0x9eb fp=0xc0045cff40 sp=0xc0045cfcf8 pc=0x354ad8b
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
        github.com/cockroachdb/cockroach/pkg/sql/planhook.go:148 +0x5b fp=0xc0045cffe0 sp=0xc0045cff40 pc=0x30d949b
runtime.goexit()
        GOROOT/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc0045cffe8 sp=0xc0045cffe0 pc=0x4bf181
created by github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec
        github.com/cockroachdb/cockroach/pkg/sql/planhook.go:147 +0x145
```

--- a/pkg/sql/catalog/dbdesc/database_desc_builder.go
+++ b/pkg/sql/catalog/dbdesc/database_desc_builder.go
@@ -96,6 +96,12 @@ func (ddb *databaseDescriptorBuilder) RunRestoreChanges(
 func maybeConvertIncompatibleDBPrivilegesToDefaultPrivileges(
 	privileges *descpb.PrivilegeDescriptor, defaultPrivileges *descpb.DefaultPrivilegeDescriptor,
 ) (hasChanged bool) {
+	// If privileges are nil, there is nothing to convert.
+	// This case can happen during restore where privileges are not yet created.
+	if privileges == nil {
+		return false
+	}
+
 	var pgIncompatibleDBPrivileges = privilege.List{
 		privilege.SELECT, privilege.INSERT, privilege.UPDATE, privilege.DELETE,
 	}

